cmd/snap-update-ns/bootstrap.go

```diff
@@ -28,9 +28,7 @@ package main
 #include <stdlib.h>
 #include "bootstrap.h"
 
-__attribute__((constructor)) static void init(void) {
-	bootstrap();
-}
+__attribute__((section(".preinit_array"), used)) static typeof(&bootstrap) init = &bootstrap;
```

Comment:
Could you please add a comment explaining how this works:
```
The bootstrap function is called by the runtime before passing control to main.
We are using the `preinit_array` instead of `init_array` in case go later on decides
to use the latter for any initialization which would defeat the purpose of the C
bootstrap code.
```